### PR TITLE
fix: generate csrf tokens on server not client

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This is part of the set of changes we need to remove `crypto` in the client.  It was never necessary, since we can always send the token out from the server as a cookie and back as a header, never needing to use `_csrf` (and `crypto` to generate tokens) on the client.

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
Although it seems to be setting the I see the 'CSRF-Server-Token' right after auth, so I assume it is working properly.

**Comment by ojeytonwilliams** [comment]
Do you mean it's setting `_csrf`?  I've investigated a little and `Session` duration leaves it up to the browser how long those cookies stick around.  With Firefox it survives page reload, but not closing down of the browser.  Brave, in contrast keeps it until you actively log out of fcc.

**Review by raisedadead — APPROVED** [review]

File: api-server/src/server/middlewares/csurf.js [hunk 1]
```diff
@@ -16,8 +18,13 @@ export default function getCsurf() {
...
       )
     ) {
-      return next();
+      next();
+    } else {
+      // add the middleware
+      protection(req, res, next);
+      // use the middleware to generate a token. The client sends this back via
+      // a header
+      res.cookie('CSRF-Server-Token', req.csrfToken(), opts);
```
Comment (raisedadead) [inline api-server/src/server/middlewares/csurf.js:27]:
```suggestion
      res.cookie('csrf_token', req.csrfToken(), opts);
```

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: client/src/utils/ajax.js [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
-// via headers:
+// CSRF-Server-Token is passed to the client as a cookie. The client must send
+// this back as a header.
 function setCSRFTokens() {
-  const _csrf = typeof window !== 'undefined' && cookies.get('_csrf');
-  if (!_csrf) return;
-  axios.defaults.headers.post['CSRF-Token'] = tokens.create(_csrf);
-  axios.defaults.headers.put['CSRF-Token'] = tokens.create(_csrf);
+  const csrfToken =
+    typeof window !== 'undefined' && cookies.get('CSRF-Server-Token');
```
Comment (raisedadead) [inline client/src/utils/ajax.js:15]:
```suggestion
    typeof window !== 'undefined' && cookies.get('csrf_token');
```

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Review by raisedadead — APPROVED** [review]

**Review by ahmaxed — APPROVED** [review]

**Comment by ahmadabdolsaheb** [comment]
Just a heads up that we should test the token in .dev before it goes to production.

**Pull request merged**